memory/persona.py

```
@@ -42,9 +42,122 @@
 AUTO_CONFIRM_DAYS = 3                # pending reflection N 天无反对 → 自动晋升
 
+# ── 负面话题回避 ─────────────────────────────────────────────────
+TOPIC_HARD_AVOID_THRESHOLD = 2       # 同一话题命中 2 次后升级为长期不提
+_GENERIC_AVOID_SIGNAL_RE = re.compile(
+    r'(?:^|[，。！？!?、,\s])(?:不要|别)(?:再)?(?P<topic>[^，。！？!?]{2,24}?)(?:了|啦|呀|啊|吧|嘛|呢|哦|喔|$)'
+)
+
+_NEGATIVE_PATTERNS = (
+    (re.compile(r'(?:别提|不要提|别再提|不要再提|不想聊|别聊|不要聊|不想提|别说|不要说)(?P<topic>.+?)(?:了|啦|呀|啊|吧|。|！|!|？|\?|,|，|$)'), True),
+    (_GENERIC_AVOID_SIGNAL_RE, True),
+    (re.compile(r'(?:不喜欢|不爱吃|讨厌吃|不想吃)(?P<topic>.+?)(?:了|啦|呀|啊|吧|嘛|呢|。|！|!|？|\?|,|，|$)'), True),
```

Ruff (0.15.9)

[warning] String contains ambiguous `！` (FULLWIDTH EXCLAMATION MARK). Did you mean `!` (EXCLAMATION MARK)?

(RUF001)